cmd/oci/oci.go

```diff
@@ -61,6 +62,15 @@ var OCICmd = &cobra.Command{
 		}
 		platform = p
 
+		if tag != "" {
+			if strings.Contains(artifact.Name, ":") {
+				parts := strings.Split(artifact.Name, ":")
+				artifact.Name = fmt.Sprintf("%s:%s", parts[0], tag)
```

Comment:
Changing the `artifact.Name` here will only affect the name when we are loading the artifact to docker daemon.

This is the bit where we are passing the name to the flake generation:
https://github.com/buildsafedev/bsf/blob/0cb31b1050f2622da13663f1206b545b5b3241dd/pkg/nix/template/oci.go#L29

and this flake is being generated from here:
https://github.com/buildsafedev/bsf/blob/0cb31b1050f2622da13663f1206b545b5b3241dd/pkg/generate/generate.go#L20

In this same function, you will see we fetch the config again by reading `bsf.hcl` which means the name will remain the same for the artifact.

What we can do is, just read the config in oci command itself and then pass it further to `generate.Generate` function over here:
https://github.com/buildsafedev/bsf/blob/0cb31b1050f2622da13663f1206b545b5b3241dd/cmd/oci/oci.go#L69
so that we have the tag in the nix flake as well.